### PR TITLE
shipit_static_analysis: Move repo_dir management in settings

### PR DESCRIPTION
**Pull request by La0**
This is a part of the old pull request #861. I'm rebasing and splitting it in several smaller chunks.

### CODE REVIEW

File: src/shipit_static_analysis/shipit_static_analysis/lint.py [hunk 1]
```diff
@@ -28,9 +29,8 @@
 
 
 class MozLintIssue(Issue):
-    def __init__(self, repo_dir, repo_path, column, level, lineno, linter, message, rule, **kwargs):
+    def __init__(self, repo_path, column, level, lineno, linter, message, rule, **kwargs):
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/lint.py]: What's repo_path?

File: src/shipit_static_analysis/shipit_static_analysis/lint.py [hunk 2]
```diff
@@ -155,14 +154,14 @@ def find_issues(self, path):
...
             logger.warn('Missing path in linter output', path=path)
             return
 
         # Mozlint uses both full & relative path to index issues
         return [
-            MozLintIssue(self.repo_dir, path, **issue)
+            MozLintIssue(path, **issue)
             for p in (path, full_path)
-            for issue in payload.get(p, [])
+            for issue in payload[full_path]
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/lint.py]: Why this change? I remember you saw mozlint issues appearing either with full paths or relative paths.

File: src/shipit_static_analysis/shipit_static_analysis/lint.py [hunk 3]
```diff
@@ -155,14 +154,14 @@ def find_issues(self, path):
...
-        full_path = os.path.join(self.repo_dir, path)
+        full_path = os.path.join(settings.repo_dir, path)
         if full_path not in payload and path not in payload:
             logger.warn('Missing path in linter output', path=path)
             return
 
         # Mozlint uses both full & relative path to index issues
         return [
-            MozLintIssue(self.repo_dir, path, **issue)
+            MozLintIssue(path, **issue)
```
Comment (marco-c) [inline src/shipit_static_analysis/shipit_static_analysis/lint.py]: Should `path` be `p`?

File: src/shipit_static_analysis/tests/conftest.py [hunk 4]
```diff
@@ -34,33 +44,37 @@ def mock_config():
     )
 
     from shipit_static_analysis.config import settings
-    settings.setup('test')
+    tempdir = tempfile.mkdtemp()
```
Comment (marco-c) [inline src/shipit_static_analysis/tests/conftest.py]: You might want to yield `settings`, so that you can remove the temporary directory.

File: src/shipit_static_analysis/shipit_static_analysis/lint.py (hunk 1 above)
Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/lint.py]: `repo_path` is the path of the file for this issue, in mozilla-central

File: src/shipit_static_analysis/shipit_static_analysis/lint.py (hunk 3 above)
Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/lint.py]: Yeap, same bad rebase

Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/lint.py]: Actually, no, it must be `path` as MozlintIssue needs the relative path.

File: src/shipit_static_analysis/shipit_static_analysis/lint.py (hunk 1 above)
Comment (marco-c) [reply src/shipit_static_analysis/shipit_static_analysis/lint.py]: Maybe `path` would be clearer, to avoid confusion

File: src/shipit_static_analysis/shipit_static_analysis/lint.py (hunk 3 above)
Comment (marco-c) [reply src/shipit_static_analysis/shipit_static_analysis/lint.py]: Right, this code is a bit confusing, maybe because of the two for loops in the comprehension. It might be worth refactoring it a bit, but we can do it later on.

Comment (La0) [reply src/shipit_static_analysis/shipit_static_analysis/lint.py]: :+1:
